components/admin/data-table/data-table.composition.tsx

```diff
@@ -2,9 +2,23 @@ import React from "react"; //
 import { DataTable } from "./data-table";
 
 const row = [
-  { name: "Duah", address: "bn-34343", title: "Ninja 3" },
-  { name: "Rahman", address: "bx-34343", title: "Ninja 5" },
-  { name: "Kojo", address: "bz-34343", title: "Ninja 4" },
+  {
+    name: "Duah",
+    address: "bn-34343",
+    title: "Ninja 3",
+    onClick: () => console.log("Duah"),
+  },
+  {
+    name: "Rahman",
+    address: "bx-34343",
+    title: "Ninja 5",
+    onClick: () => console.log("Rahman"),
```

Comment:
Why is the click defined on the data? If I load data from the server how will it have a click?

Reply:
If we want to assign a click event to it, I think the best place to do that is before passing the data to the component. That way, we can control what the click even does without having to change the component at all.

Reply:
Is the user clicking the table or the data? If we have 2 tables using the same data we add 2 functions on the data right?

Reply:
The data

Reply:
The data is not a UI element so how does a user click it? The user can only click a UI element that represents the data.
Assuming we have 2 tables showing same data and different format.

But each table does something different when the row is clicked. How would you handle that if the onClick is defined on the data?